Annotation class not available: @Any

## Describe the bug
When migrating from Quarkus `1.13.7` to `2.0.0` I began getting the below exception (referencing the `javax.enterprise.inject.Any`). Unfortunately I have not been able to isolate the problematic code from my baseline to create a reproducer. I am open to any suggestions on how to fix this.

### Expected behavior
Successful build

### Actual behavior
```
Caused by: java.lang.NullPointerException: Annotation class not available: @Any
        at io.quarkus.arc.processor.AnnotationLiteralProcessor.process(AnnotationLiteralProcessor.java:61)
        at io.quarkus.arc.processor.DecoratorGenerator.createConstructor(DecoratorGenerator.java:282)
        at io.quarkus.arc.processor.DecoratorGenerator.generate(DecoratorGenerator.java:103)
        at io.quarkus.arc.processor.BeanProcessor.generateResources(BeanProcessor.java:183)
        at io.quarkus.arc.deployment.ArcProcessor.generateResources(ArcProcessor.java:457)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:820)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2442)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1476)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```


## To Reproduce

I haven't been able to isolate a reproducer from our baseline.

## Environment (please complete the following information):

### Output of `uname -a` or `ver`
MINGW64_NT-10.0-18363

### Output of `java -version`
OpenJDK 11.0.11

### Quarkus version or git rev
2.0.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)
Gradle 6.9

--- a/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/DecoratorGenerator.java
+++ b/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/DecoratorGenerator.java
@@ -277,7 +277,7 @@ public class DecoratorGenerator extends BeanGenerator {
             ResultHandle delegateQualifiersHandle = constructor.newInstance(MethodDescriptor.ofConstructor(HashSet.class));
             for (AnnotationInstance delegateQualifier : decorator.getDelegateQualifiers()) {
                 // Create annotation literal first
-                ClassInfo delegateQualifierClass = decorator.getDeployment().getInterceptorBinding(delegateQualifier.name());
+                ClassInfo delegateQualifierClass = decorator.getDeployment().getQualifier(delegateQualifier.name());
                 constructor.invokeInterfaceMethod(MethodDescriptors.SET_ADD, delegateQualifiersHandle,
                         annotationLiterals.process(constructor, classOutput, delegateQualifierClass, delegateQualifier,
                                 Types.getPackageName(creator.getClassName())));

--- a/independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/decorators/qualifiers/DelegateQualifiersTest.java
+++ b/independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/decorators/qualifiers/DelegateQualifiersTest.java
@@ -1,0 +1,72 @@
+package io.quarkus.arc.test.decorators.qualifiers;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.test.ArcTestContainer;
+import io.quarkus.arc.test.MyQualifier;
+import javax.annotation.Priority;
+import javax.decorator.Decorator;
+import javax.decorator.Delegate;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.Dependent;
+import javax.enterprise.util.AnnotationLiteral;
+import javax.inject.Inject;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class DelegateQualifiersTest {
+
+    @RegisterExtension
+    public ArcTestContainer container = new ArcTestContainer(Converter.class, ToUpperCaseConverter.class,
+            TrimConverterDecorator.class, MyQualifier.class);
+
+    @Test
+    public void testDecoration() {
+        @SuppressWarnings("serial")
+        ToUpperCaseConverter converter = Arc.container()
+                .instance(ToUpperCaseConverter.class, new AnnotationLiteral<MyQualifier>() {
+                }).get();
+        assertEquals("HOLA!", converter.convert(" holA!"));
+        assertEquals(" HOLA!", converter.convertNoDelegation(" holA!"));
+    }
+
+    interface Converter<T> {
+
+        T convert(T value);
+
+    }
+
+    @MyQualifier
+    @ApplicationScoped
+    static class ToUpperCaseConverter implements Converter<String> {
+
+        @Override
+        public String convert(String value) {
+            return value.toUpperCase();
+        }
+
+        public String convertNoDelegation(String value) {
+            return value.toUpperCase();
+        }
+
+    }
+
+    @Dependent
+    @Priority(1)
+    @Decorator
+    static class TrimConverterDecorator implements Converter<String> {
+
+        @Inject
+        @MyQualifier
+        @Delegate
+        Converter<String> delegate;
+
+        @Override
+        public String convert(String value) {
+            return delegate.convert(value.trim());
+        }
+
+    }
+
+}

